konduit-serving-api/src/main/java/ai/konduit/serving/pipeline/PipelineStep.java

```diff
@@ -27,6 +30,50 @@
 public interface PipelineStep extends Serializable {
 
 
+    /**
+     * Returns true if the input data format
+     * is valid or not
+     * @param dataFormat the dataformat to test
+     * @return true if the input format is valid or false otherwise
+     */
+   default  boolean isValidInputType(Input.DataFormat dataFormat) {
+     if(validInputTypes() == null || validInputTypes().length < 1) {
+         return true;
+     }
+     return Arrays.stream(validInputTypes()).anyMatch(input -> dataFormat.equals(input));
+   }
+
+
+    /**
+     * Returns true if the output data format
+     * is valid or not
+     * @param dataFormat the dataformat to test
+     * @return true if the output format is valid or false otherwise
+     */
+   default  boolean isValidOutputType(Output.DataFormat dataFormat) {
+       if(validOutputTypes() == null || validOutputTypes().length < 1) {
+           return true;
+       }
+
+       return Arrays.stream(validOutputTypes()).anyMatch(input -> dataFormat.equals(input));
+
+   }
+
+
+    /**
+     * Returns the valid {@link Input.DataFormat}
+     * when the input is the beginning of a pipeline
+     * @return
```

Comment:
maybe add a note that explains that setting this to `null` is the intended way of allowing any format (for both inputs and outputs)?